### DISCUSSION TITLE
"Pseudoproxy" repo? Remote download of artifacts through a middle man?

### DISCUSSION THREAD

**Discussion by michael-etzkorn** [sbt/sbt#6634] [Q&A] [answered]
I'm new to `sbt` and have a gate server that connects to a local CAD server. In order to make a remote connection, the local server cannot have an internet connection other than to the gate server (as far as I'm aware). The gate server can resolve dependencies using sbt, but I'm wondering if there's a way to have the gate server fetch the dependencies and pass them to the local server without installing a full blown proxy server (installation would require getting IT involved).

**Comment by eed3si9n** [comment]
I'm not sure if that qualifies as full-blown proxy server, but JFrog makes Artifactory, which does what you're describing, and it was fairly common among sbt developers to have an instance of Artifactory available either on a laptop or a home server if/when we wanted to avoid hitting the internet for some testing.

**Comment by michael-etzkorn** [comment]
Artifactory still seems more complicated than what I was hoping for. I've managed to get on these past few months by just connecting the local server to the internet every now and then when the `build.sbt` gets updated, but it's a personal solution, and I can't expand it well to even my small team. I guess I'm hoping to get my hand held for how to find and move the jars and files sbt downloads from the gate server to the local server. Or between local servers. Or from one user's home folder to another user's home folder. And then make those jar files available to the sbt client that's in offline mode on the local server. 

I'm glad I suggested Scala because it's made designing really productive. However, I've encountered issues with having other engineers run my project. I'd ideally just put everything in a shared public place because everything _is_ already downloaded somewhere, it just needs to be made seen by whoever's sbt is launching.

**Reply by eed3si9n** [reply]
Maybe you can try `mvn install` to install sbt's artifacts to a local Maven repo directory, and then you can move the directory to a shared drive or host it as HTTP, as lightweight Artifactory.

**Reply by er1c** [reply]
If you want to use a public "artifactory"  you can try https://github.com/er1c/sbt-github

**Comment by michael-etzkorn** [comment]
I overcomplicated this.  

I believe I can use the gate server as an http / socks proxy for when building for the first time (which is when sbt is making http/https requests) and then switch to offline mode once the artifacts are downloaded.

**Comment by michael-etzkorn** [comment, marked as answer]
Yeah sorry if "full blown proxy server" was misleading, I sort of A and B problemed myself with ignorance. I just didn't want to have to install anything. 

I added `-DsocksProxyHost=mySocksServer -DsocksProxyPort=myProxyPort` to my `$JAVA_OPTS` variable. 

This solution has worked for me: 
https://stackoverflow.com/questions/25445875/how-to-set-nonproxyhosts-for-a-socks-proxy
